### ISSUE TITLE
False positive finding for GENERIC_VARIABLE_WRONG_DECLARATION rule

### ISSUE THREAD

**Issue by WebTiger89** [saveourtool/diktat#1317]
I don't know what the problem is. It might be a false positive warning.

The suspicious code snippet:
```
inline fun <K, V> Iterable<K>.associateWithNotNull(valueSelector: (K) -> V?): Map<K, V> {
    val result = LinkedHashMap<K, V>(
        mapCapacity(collectionSizeOrDefault(10)).coerceAtLeast(16),
    )
    return associateWithNotNullTo(result, valueSelector)
}
```

**Comment by akuleshov7** [comment]
`result` variable should have explicit type declaration, because your generic is complex. That's it :)

**Comment by akuleshov7** [comment]
In Kotlin no need to use constructor directly and create two maps. There are utility methods that can be useful to create a map, for example: `val result = linkedMapOf(1 to 2)`

**Comment by WebTiger89** [comment]
Damn my bad. I should have check the config file description in this case. This function is almost a  1:1 copy from android source code. Exceptionally this code snippet is correct as is. After adding the explicit type the warning is gone. Thanks for the good explanation tho :)
